File: src/main/kotlin/ex04/Exam04.java
package ex04;

import java.util.HashMap;
import java.util.Map;

public class Exam04 {

    public Exam04(){
        var hashMap = new HashMap<String, Object>();
        hashMap.put("key", "value");
        var map = Map.of("key1", "value1", "ke2", "value2");
        map.put("key3", "value3");

        hashMap.get("key");
    }


    public static void main(String[] args){
        new Exam04();
    }
}

File: src/main/kotlin/ex04/Exam04.kt
package ex04

fun main(){

    // java Object -> Any
    // 불면 map
    val map = mapOf<String, Any>(
        Pair("key1", "value1"),
        "key2" to "value2"
    )

    // 가변 mutalbleMap
    val mutalbleMap = mutableMapOf(
        "key1" to "value1" // 타입추론
    )

    mutalbleMap.put("key1", "value1")
    val value1 = mutalbleMap.get("key1")

    mutalbleMap["key2"] = "value2"
    val value2 = mutalbleMap["key2"]

    // hashMapOf
    val hashMap = hashMapOf(
        "key1" to "value1"
    )

    // 값을 3가지 가지는 객체
    var triple = Triple<String, String, String>(
        first = "",
        second = "",
        third = ""
    )
}
